llama-index-integrations/llms/llama-index-llms-githubllm/llama_index/llms/githubllm/base.py

```diff
@@ -0,0 +1,296 @@
+import os
+import logging
+import time
+from typing import Any, Dict, List, Sequence, Generator
+
+import requests
+from llama_index.core.base.llms.types import (
+    ChatMessage,
+    ChatResponse,
+    ChatResponseGen,
+    CompletionResponse,
+    CompletionResponseGen,
+    LLMMetadata,
+)
+from llama_index.core.llms.callbacks import llm_chat_callback, llm_completion_callback
+from llama_index.core.llms.custom import CustomLLM
+from llama_index.core.bridge.pydantic import Field, PrivateAttr, validator
+
+logger = logging.getLogger(__name__)
+
+
+class GithubLLM(CustomLLM):
```

Comment:
This looks pretty good to me. Do we want to implement async endpoints at all? Could be in another PR as well, up to you

Reply:
thanks for the kind words. for async , i think there's a bit more thinking to do though. it could be done in this PR or in another where multimodal and embedding models are also added ?

Reply:
I think async here would just be using async network requests. But we can definitely do it in another PR :)